jetty-server/src/test/java/org/eclipse/jetty/server/handler/FileBufferedResponseHandlerTest.java

```diff
@@ -0,0 +1,307 @@
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.toolchain.test.FS;
+import org.eclipse.jetty.toolchain.test.MavenTestingUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+/**
+ * Resource Handler test
+ */
+public class FileBufferedResponseHandlerTest
+{
+    private static Server _server;
+    private static LocalConnector _local;
+    private static TestHandler _test;
+    private static Path _testDir;
+
+    @BeforeAll
+    public static void setUp() throws Exception
+    {
+        _testDir = MavenTestingUtils.getTargetTestingPath(FileBufferedResponseHandlerTest.class.getName());
+        FS.ensureDirExists(_testDir);
+
+        _server = new Server();
+        HttpConfiguration config = new HttpConfiguration();
+        config.setOutputBufferSize(1024);
+        config.setOutputAggregationSize(256);
+        _local = new LocalConnector(_server, new HttpConnectionFactory(config));
+        _server.addConnector(_local);
+
+        FileBufferedResponseHandler bufferedHandler = new FileBufferedResponseHandler();
+        bufferedHandler.setTempDir(_testDir);
+        bufferedHandler.getPathIncludeExclude().include("/include/*");
+        bufferedHandler.getPathIncludeExclude().exclude("*.exclude");
+        bufferedHandler.getMimeIncludeExclude().exclude("text/excluded");
+        bufferedHandler.setHandler(_test = new TestHandler());
+
+        ContextHandler contextHandler = new ContextHandler("/ctx");
+        contextHandler.setHandler(bufferedHandler);
+
+        _server.setHandler(contextHandler);
+        _server.start();
+    }
+
+    @AfterAll
+    public static void tearDown() throws Exception
+    {
+        _server.stop();
+    }
+
+    @BeforeEach
+    public void before()
+    {
+        FS.ensureEmpty(_testDir);
+        _test._bufferSize = -1;
+        _test._mimeType = null;
+        _test._content = new byte[128];
+        Arrays.fill(_test._content, (byte)'X');
+        _test._content[_test._content.length - 1] = '\n';
+        _test._writes = 10;
+        _test._flush = false;
+        _test._close = false;
+        _test._reset = false;
+    }
+
+    public static int getNumFiles()
+    {
+        File[] files = _testDir.toFile().listFiles();
+        if (files == null)
+            return 0;
+
+        return files.length;
+    }
+
+    @Test
+    public void testPathNotIncluded() throws Exception
+    {
+        String response = _local.getResponse("GET /ctx/path HTTP/1.1\r\nHost: localhost\r\n\r\n");
+        assertThat(response, containsString(" 200 OK"));
+        assertThat(response, containsString("Write: 0"));
+        assertThat(response, containsString("Write: 7"));
```

Comment:
Sorry, I failed to understand this test.
Where does the number `7` come from? 
You should not rely on "oh, on this version of this JVM of this vendor, on my current computer, the number of writes is going to be 8 before ???".
What happens at write number 9?